### PR TITLE
Make ServiceBusReceivedMessage immutable

### LINKED ISSUE #11782: [QUERY] ServiceBusReceivedMessage
**Query/Question**
1. Is `ServiceBusReceivedMessage` supposed to be immutable?
1. How a custom testing code that requires `ServiceBusReceivedMessage` would obtain the object w/o reflection?

**Environment:**
 - Name and version of the Library package used: Azure.Messaging.ServiceBus/7.0.0-preview.2
 - Hosting platform or OS and .NET runtime version Windows 10 .NET / Framework 4.8 / Core 3.1
 - IDE and version : Visual Studio 2019

### PR DESCRIPTION
**Pull request by SeanFeldman**
Fixes #11782

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Azure.Messaging.ServiceBus.csproj [hunk 1]
```diff
@@ -18,6 +18,7 @@
     <PackageReference Include="Azure.Identity" />
     <PackageReference Include="Microsoft.Azure.Amqp" />
     <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" />
+    <PackageReference Include="System.Collections.Immutable" />
```
Comment (JoshLove-msft) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Azure.Messaging.ServiceBus.csproj:21]: I don't see anywhere this is actually used. Did you forget to commit ServiceBusReceivedMessage?

Comment (SeanFeldman) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Azure.Messaging.ServiceBus.csproj:21]: As a friend of mine used to say - sucks to be an idiot. Sorry, updated.

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Azure.Messaging.ServiceBus.csproj:21]: > As a friend of mine used to say - sucks to be an idiot. Sorry, updated.

Hehe, he doesn't say it anymore?

Comment (SeanFeldman) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Azure.Messaging.ServiceBus.csproj:21]: Nope. He's no longer around to say that.

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Azure.Messaging.ServiceBus.csproj:21]: Ah, sorry about that.

**Review by JoshLove-msft — APPROVED** [review]
LGTM - will merge assuming the live test run is all green. Thanks @SeanFeldman

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Azure.Messaging.ServiceBus.csproj (hunk 1 above)
Comment (SeanFeldman) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Azure.Messaging.ServiceBus.csproj:21]: All good. Just don't continue his tradition 🙂

**Pull request merged**